server/src/main/java/io/crate/types/DataType.java

```diff
@@ -100,7 +100,7 @@ public enum Precedence {
      * type should be preferred (higher precedence) or converted (lower
      * precedence) during type conversions.
      */
-    public abstract Precedence precedence();
+    protected abstract Precedence precedence();
```

Comment:
Made this protected to ensure consumers go through `precedes` instead of checking precedence manually and missing part of the logic.